synapse/handlers/room_member.py

```diff
@@ -400,11 +400,13 @@ async def _local_membership_update(
         # do this check just before we persist an event as well, but may as well
         # do it up front for efficiency.)
         if txn_id and requester.access_token_id:
-            existing_event_id = await self.store.get_event_id_from_transaction_id(
-                room_id,
-                requester.user.to_string(),
-                requester.access_token_id,
-                txn_id,
+            existing_event_id = (
+                await self.store.get_event_id_from_transaction_id_and_token_id(
+                    room_id,
+                    requester.user.to_string(),
+                    requester.access_token_id,
+                    txn_id,
+                )
```

Comment:
Do we not also need to check the new table here?

Reply:
Good catch, fixed it